### DISCUSSION TITLE
Error connecting to the Mongo database, invalid connection string

### DISCUSSION THREAD

**Discussion by WebTerminator** [strapi/strapi#5865] [Legacy Topics]
**Describe the bug**
I am trying to run a production env locally (after having built my portfolio with this [strapi tuorial](https://strapi.io/blog/strapi-starter-next-js-blog)) to test it before pushing to Heroku.
I have followed [these instructions](https://strapi.io/documentation/3.0.0-beta.x/guides/databases.html#mongodb-installation) but when I run `NODE_ENV=production yarn start` within the backend project I get: 

![Screen Shot 2020-04-18 at 13 40 35](https://user-images.githubusercontent.com/4947347/79637861-47ded300-817a-11ea-96f2-50ebf90ac1a4.png)

**Steps to reproduce the behavior**

1. cd backend
2. NODE_ENV=production yarn start

**Expected behavior**
Successful production environment running

**Code snippets**
This is my production database and the variables have been set in heroku as per [these instructions](https://strapi.io/documentation/3.0.0-beta.x/deployment/heroku.html)

```
heroku config:set DATABASE_URI="mongodb://paulbocuse:<password>@strapidatabase-shard-00-00-fxxx6c.mongodb.net:27017,strapidatabase-shard-00-01-fxxxc.mongodb.net:27017,strapidatabase-shard-00-02-fxxxc.mongodb.net:27017/test?ssl=true&replicaSet=strapidatabase-shard-0&authSource=admin&retryWrites=true&w=majority"
heroku config:set DATABASE_NAME="my-database-name"
```

```
{
  "defaultConnection": "default",
  "connections": {
    "default": {
      "connector": "mongoose",
      "settings": {
        "uri": "${process.env.DATABASE_URI}",
        "database": "${process.env.DATABASE_NAME}"
      },
      "options": {
        "ssl": true
      }
    }
  }
}
```
backend package.json
```
...
  "dependencies": {
    "fs": "^0.0.1-security",
    "knex": "<0.20.0",
    "path": "^0.12.7",
    "rimraf": "^3.0.1",
    "slugify": "^1.4.0",
    "sqlite3": "latest",
    "strapi": "3.0.0-beta.18.4",
    "strapi-admin": "3.0.0-beta.18.4",
    "strapi-connector-bookshelf": "^3.0.0-beta.20",
    "strapi-connector-mongoose": "^3.0.0-beta.20",
    "strapi-plugin-content-manager": "3.0.0-beta.18.4",
    "strapi-plugin-content-type-builder": "3.0.0-beta.18.4",
    "strapi-plugin-email": "3.0.0-beta.18.4",
    "strapi-plugin-graphql": "3.0.0-beta.18.4",
    "strapi-plugin-upload": "3.0.0-beta.18.4",
    "strapi-plugin-users-permissions": "3.0.0-beta.18.4",
    "strapi-utils": "3.0.0-beta.18.4",
    "unzip-stream": "^0.3.0"
  },
...
```

**System**

- Node.js version: 10.0.0
- NPM version: 5.6.0
- Operating system: macOS High Sierra 10.13.6

**Comment by derrickmehaffy** [comment]
`Strapi version: 3.0.0-alpha.26.2` Alpha is no longer supported you should update to the beta.

**Comment by WebTerminator** [comment]
@derrickmehaffy that was my global strapi but the one I use for the project is beta as uploaded above. Thanks

**Comment by derrickmehaffy** [comment]
You are mixing different versions of Strapi, this will need to be corrected

**Comment by derrickmehaffy** [comment]
@WebTerminator I've moved this to a discussion as it's not a bug with Strapi but your own settings that are causing the failure

**Reply by derrickmehaffy** [reply]
See the following example for the database.json on mongo atlas: https://gist.github.com/derrickmehaffy/aea3c635e0b9704b14fd1693fc56fea2

**Comment by lauriejim** [comment]
Here is a [video tuto](https://www.youtube.com/watch?v=HHnRsZwFmR8&list=PL7Q0DQYATmvhlHxHqfKHsr-zFls2mIVTi&index=7&t=0s) that shows you how to deal with that I think.

**Reply by abaze** [reply]
I had the same issue but I solved it very simply, in the doc you we found this : 
` "host": "${process.env.DATABASE_HOST}",`
You have to replace the double quote " by ` (ES6 syntax)
